Allow users to configure webhook message content with a template file

supasteev0 opened the pull request:
Description of changes:

When the destination of the webhook requires a more complex message content, it can be easier to provide a json template file which will be mounted into the pod, rather than having the webhook message content set in an inline variable.

This will allow users to set a WEBHOOK_TEMPLATE_FILE environment variable (--webhook-template-file argument) to set the template file path which will be used to send the webhook message. This takes precedence over the --webhook-template argument.

Template file will have to exist (be mounted into the pod as volume by users) otherwise it will throw an error.

Example: 
```
---
apiVersion: v1
data:
  webhook-alert-template.json: |-
    {
      "status": "firing",
      "alerts": [
          {
            "status": "firing",
            "labels": {
              "alertname": "myalert",
              "region": "eu-west-1",
              "severity": "high"
            }
          }
       ]
    }
kind: ConfigMap
metadata:
  name: webhook-template
  namespace: monitoring
---
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    app: aws-node-termination-handler
  name: aws-node-termination-handler
  namespace: monitoring
spec:
  selector:
    matchLabels:
      app: aws-node-termination-handler
  template:
    metadata:
      labels:
        app: aws-node-termination-handler
    spec:
      containers:
      - env:
        ...
        - name: WEBHOOK_TEMPLATE_FILE
          value: /config/webhook-alert-template.json
        ...
        volumeMounts:
        - mountPath: /config/
          name: template
      volumes:
      - configMap:
          name: webhook-template
        name: template
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

bwagner5 reviewed (changes requested):
Rather than making the user mount the file, can you instead use a ConfigMap similar to the way the `webhookURLSecretName` works.

supasteev0 commented:
@bwagner5 I have update the Helm chart by adding 2 new variables in order to set the configmap name and the template file name that will be mounted into the pod.

bwagner5 commented on pkg/webhook/webhook.go:
```diff
@@ -35,8 +36,21 @@ type combinedDrainData struct {
+			return
+		}
+		webhookTemplateContent = string(content)
+		log.Log().Msgf("Template file content - %s", webhookTemplateContent)
```
can you move this to`log.Debug() .Msgf(`

supasteev0 replied:
Done

bwagner5 reviewed (changes requested):
This looks good! I missed one thing during my first pass review.

Can you update the `ValidateWebhookConfig()` function as well? The validate function is called on startup so you know if your template is invalid upfront rather than waiting for a termination event for the error to be thrown. 

https://github.com/aws/aws-node-termination-handler/blob/master/pkg/webhook/webhook.go#L103

supasteev0 commented:
I updated the ValidateWebhookConfig function as requested. I also fixed a typo on the nthConfig variable which was written in 2 different ways. Looks like the travis test has failed because of an issue with ec2Mock not responding, but I don't know how to re trigger the job.

bwagner5 reviewed (approved):
LGTM! 🚀 thanks for the contribution!
